src/main/java/io/jenkins/plugins/pipeline_elasticsearch_logs/write/fluentd/FluentdWriter.java

```diff
@@ -0,0 +1,268 @@
+    private int bufferCapacity = DEFAULT_BUFFER_CAPACITY;
+
+    @DataBoundConstructor
+    public FluentdWriter() throws URISyntaxException {
+    }
+
+    public String getTagPrefix() {
+        return tagPrefix;
+    }
+
+    @DataBoundSetter
+    public void setTagPrefix(String tagPrefix) {
+        this.tagPrefix = tagPrefix;
+        this.configChanged = true;
+    }
+
+    public String getTag() {
+        return tag;
+    }
+
+    @DataBoundSetter
+    public void setTag(String tag) {
+        this.tag = tag;
+        this.configChanged = true;
+    }
+
+    public String getHost() {
+        return host;
+    }
+
+    @DataBoundSetter
+    public void setHost(String host) {
+        this.host = host;
+        this.configChanged = true;
+    }
+
+    public int getPort() {
+        return port;
+    }
+
+    @DataBoundSetter
+    public void setPort(int port) {
+        if(port < 0 || port > 65535) throw new IllegalArgumentException("Port out of range: " + port);
+        this.port = port;
+        this.configChanged = true;
+    }
+
+    public int getRetryMillis() {
+        return retryMillis;
+    }
+
+    @DataBoundSetter
+    public void setRetryMillis(int retryMillis) {
+        if (retryMillis < 0)
+            retryMillis = 0;
+        this.retryMillis = retryMillis;
+        this.configChanged = true;
+    }
+
+    public int getTimeoutMillis() {
+        return timeoutMillis;
+    }
+
+    @DataBoundSetter
+    public void setTimeoutMillis(int timeoutMillis) {
+        if(timeoutMillis < 0) throw new IllegalArgumentException("timeoutMillis less than 0");
+        this.timeoutMillis = timeoutMillis;
+        this.configChanged = true;
+    }
+
+    public int getBufferCapacity() {
+        return bufferCapacity;
+    }
+
+    @DataBoundSetter
+    public void setBufferCapacity(int bufferCapacity) {
+        if (bufferCapacity < 0)
+            bufferCapacity = DEFAULT_BUFFER_CAPACITY;
+        this.bufferCapacity = bufferCapacity;
+        this.configChanged = true;
+    }
+
+
+    @Extension
+    @Symbol("fluentd")
+    public static class DescriptorImpl extends ElasticSearchWriteAccessDescriptor {
+        @Override
+        public String getDisplayName() {
+            return "Fluentd Elasticsearch Writer";
+        }
+    }
+
+    /**
+     * Logs the given object to Fluentd. This method blocks if Fluentd is not reachable and
+     * throws an IOException after TIMEOUT_MILLIS.
+     *
+     * @param data The data to post
+     * @throws IOException if something failed permanently
+     */
+    @Override
+    public void push(Map<String, Object> data) throws IOException {
+        if(fluentd == null || configChanged) initFluentdLogger();
+        try {
+
+            long timeMarker = System.currentTimeMillis();
+            boolean sent = fluentd.log(tag, data);
+            if(!sent) {
+                //Errors are not reported explicitly. But the latest error should be the reason for (sent == false)
```

Comment:
fuzzy, but ok... does not work more precisely.

Reply:
Actually when sent = false it most likely means there was a severe problem, like the data is too big for the buffer.
Usually it also means the the errorhandler does not contain any information as the sender only writes to the errorhandler when writing to the socket fails.